src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+                # Note: that due to Julia starting indexing at 1, whereas Hairer
+                # indices his variable for order, k, from 0, we have
+                # k+1 = steporder in what follows
+
+    # initialization of arrays used in estimation y_np1 given dy[n],y[n], dt and
+    # list order timestep are  initialized to a maxorder x maxorder dimension
+    c, g, b, ϕstar, ϕ = intialize_coef_arrays(max_order+3, Tdy, Et, y0)
+
+    if max_order <= 13
+        γ_star = γ_star_array
+    else
+        γ_star= gamma_star_constr(max_order+3)
+    end
+
+    # Declaration of timeout variables
+    timeout_VS_const = 0 # How many steps before a variable step size can be taken
+    timeout_VS = timeout_VS_const # `timeout_VS` is a counter for this in the loop
+
+    timeout_VO_const = 2 # How many steps before a variable step order can be taken
+    timeout_VO = timeout_VO_const # `timeout_VO` is a counter for this in the loop
+
+    consec_const_steps = 0 # counts how many consecutives steps have been taken
+                           # with same step size
+
+    # Declaration of  four local error (LE) or truncation errors necessary
+    # for variable order calculations, which correspond to infinity norms of:
+    #      LE(n+1)_{k-1}, LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2}
+    # from Hairer pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0, 0.0, 0.0, 0.0
+
+    #-##########################################################################
+    # Variable Step Variable Order (VSVO) Step Integration Loop
+    #-##########################################################################
+    while  tfinal*tdir > t[n]*tdir
+        #--#####################################################################
+        # Attempt a step with current trial stepsize `dt`
+        #--#####################################################################
+        successful_step = false
+        while !successful_step
+            #---################################################################
+            # Make an approriate time step with current trial stepsize `dt`
+            #---################################################################
+            #Make the step
+            t_np1 = t[n]+dt
+            #---################################################################
+            # Calculating explicit approximatiion to y_np1 with current step dt
+            #---################################################################
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g, c, dt, t, t_np1, n, min(n, steporder+2))
+
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar, dy[n], t, t_np1, n, 1:min(max(n-1, 1), steporder+1))
+            #Note that  1<= min(max(n-1,1),steporder+1) <= n-1 and steporder+1
+
+            # Calculate next value, y_np1, with order = k = steporder + 1. See
```

Comment:
Yes, I think that will more easily explain what is going on in a first read through of the code.